bindings/python/convert.py

```diff
@@ -155,79 +215,10 @@ def create_diff(pt_infos: Dict[str, List[str]], sf_infos: Dict[str, List[str]])
-    config = AutoConfig.from_pretrained(folder)
-
-    import transformers
-
-    class_ = getattr(transformers, config.architectures[0])
-    (pt_model, pt_infos) = class_.from_pretrained(folder, output_loading_info=True)
-    (sf_model, sf_infos) = class_.from_pretrained(folder, output_loading_info=True)
-
-    if pt_infos != sf_infos:
-        error_string = create_diff(pt_infos, sf_infos)
-        raise ValueError(f"Different infos when reloading the model: {error_string}")
-
-    pt_params = pt_model.state_dict()
-    sf_params = sf_model.state_dict()
-
-    pt_shared = shared_pointers(pt_params)
-    sf_shared = shared_pointers(sf_params)
-    if pt_shared != sf_shared:
-        raise RuntimeError("The reconstructed model is wrong, shared tensors are different {shared_pt} != {shared_tf}")
-
-    sig = signature(pt_model.forward)
-    input_ids = torch.arange(10).unsqueeze(0)
-    pixel_values = torch.randn(1, 3, 224, 224)
-    input_values = torch.arange(1000).float().unsqueeze(0)
-    # Hardcoded for whisper basically
-    input_features = torch.zeros((1, 80, 3000))
-    kwargs = {}
-    if "input_ids" in sig.parameters:
-        kwargs["input_ids"] = input_ids
-    if "input_features" in sig.parameters:
-        kwargs["input_features"] = input_features
-    if "decoder_input_ids" in sig.parameters:
-        kwargs["decoder_input_ids"] = input_ids
-    if "pixel_values" in sig.parameters:
-        kwargs["pixel_values"] = pixel_values
-    if "input_values" in sig.parameters:
-        kwargs["input_values"] = input_values
-    if "bbox" in sig.parameters:
-        kwargs["bbox"] = torch.zeros((1, 10, 4)).long()
-    if "image" in sig.parameters:
-        kwargs["image"] = pixel_values
-
-    if torch.cuda.is_available():
-        pt_model = pt_model.cuda()
-        sf_model = sf_model.cuda()
-        kwargs = {k: v.cuda() for k, v in kwargs.items()}
-
-    try:
-        pt_logits = pt_model(**kwargs)[0]
-    except Exception as e:
-        try:
-            # Musicgen special exception.
-            decoder_input_ids = torch.ones((input_ids.shape[0] * pt_model.decoder.num_codebooks, 1), dtype=torch.long)
-            if torch.cuda.is_available():
-                decoder_input_ids = decoder_input_ids.cuda()
-
-            kwargs["decoder_input_ids"] = decoder_input_ids
-            pt_logits = pt_model(**kwargs)[0]
-        except Exception:
-            raise e
-    sf_logits = sf_model(**kwargs)[0]
-
-    torch.testing.assert_close(sf_logits, pt_logits)
-    print(f"Model {model_id} is ok !")
-
-
-def previous_pr(api: "HfApi", model_id: str, pr_title: str) -> Optional["Discussion"]:
+def previous_pr(api: "HfApi", model_id: str, pr_title: str, revision=Optional[str]) -> Optional["Discussion"]:
     try:
-        main_commit = api.list_repo_commits(model_id)[0].commit_id
-        discussions = api.get_repo_discussions(repo_id=model_id)
+        main_commit = api.list_repo_commits(model_id, revision=revision)[0].commit_id
```

Comment:
no need to list all commit history to get the last one. This information is available as `model_info(...).sha`


(nit) I would also rename the variable to something like `revision_commit` instead of `main_commit` (since pulling from revision and not `main`)

```suggestion
        revision_commit = api.model_info(model_id, revision=revision).sha
```

Reply:
Agreed ! (Made send before only adding the revision)